sdk/jobs/pipelines/3a_pipeline_with_spark/spark-in-pipeline.ipynb

```diff
@@ -0,0 +1,144 @@
+    "except Exception as ex:\n",
+    "    # Fall back to InteractiveBrowserCredential in case DefaultAzureCredential not work\n",
+    "    credential = InteractiveBrowserCredential()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Get a handle to workspace\n",
+    "ml_client = MLClient.from_config(credential=credential)\n",
+    "\n",
+    "# Retrieve an already attached Azure Machine Learning Compute.\n",
+    "synapse_compute = \"synapse-compute\"\n",
+    "print(ml_client.compute.get(synapse_compute))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Define pipeline\n",
+    "@pipeline(description=\"Spark pipeline job\",)\n",
+    "def spark_pipeline_job(input_data, output_data):\n",
+    "    # define the spark task\n",
+    "    spark_node = spark(\n",
+    "        code=\"./src\",\n",
+    "        entry=\"program.py\",\n",
+    "        py_files=['my_python.py'],\n",
+    "        jars=['my_java.jar'],\n",
+    "        files=['my_file.txt'],\n",
+    "        archives=['my_archive.tar'],\n",
+    "        conf={\n",
+    "            \"spark.driver.cores\": 1,\n",
+    "            \"spark.driver.memory\": \"2g\",\n",
+    "            \"spark.executor.cores\": 2,\n",
+    "            \"spark.exeutor.memory\": \"2g\",\n",
+    "            \"spark.executor.instances\": 4\n",
+    "        },\n",
+    "        environment='myenv:v1',\n",
+    "        # Alternatively, provide 'conda_file' for environment\n",
+    "        # environment={\"conda_file\": \"./conda.config\"},\n",
+    "        inputs=dict(input_folder=input_data),\n",
+    "        outputs=dict(output_folder=output_data),\n",
+    "        args=\"--myinput ${{inputs.input_folder}} --myoutput ${{outputs.output_folder}}\",\n",
+    "        # For HOBO spark, provide 'resources'\n",
+    "        # resources={\"instance_type\": \"Standard_E8S_V3\", \"runtime_version\": \"3.1.0\"}\n",
+    "    )\n",
+    "    return {\"output_file\": spark_node.outputs.output_folder}\n",
+    "\n",
+    "\n",
+    "pipeline = spark_pipeline_job(\n",
+    "    input_data=Input(path=\"azureml:mytable2:3.0\", type=\"mltable\", mode=\"direct\"),\n",
+    "    output_data=Output(type=\"uri_folder\", mode=\"direct\", path=\"wasbs://mycontainer@mystorageaccount.blob.core.windows.net/path/to/folder\"))\n",
+    "# No compute needed for HOBO spark\n",
+    "pipeline.settings.default_compute = synapse_compute"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from azure.ml.entities import load_component\n",
+    "\n",
+    "# Define pipeline using yaml\n",
+    "@pipeline(description=\"Spark pipeline job\",)\n",
+    "def spark_pipeline_job_from_yaml(input_data):\n",
+    "    # define the spark task\n",
```

Comment:
component is working as something like below:
spark_func = load_component('spark-component.yaml')
spark_node = spark_func(input1 = Input())

Reply:
Ok - one question though: how would `input1` bind to inputs if that name was both used as an input and an output? I am assuming there's no restriction sharing same name between inputs/outputs, right? Something like this:
```
inputs:
  input1:
    type: mltable
outputs:
  input1:
    type: uri_folder
    mode: direct
```
For now I simplified it by fully binding the input/outputs in the yaml file, i.e., specify the path over there.

Reply:
Closing since looks like only inputs can be passed in the partial function.